### PR TITLE
feat: can import geojson files as layer with query params

### PR DESCRIPTION
**Pull request by f-necas**
## Description
PR used to import GeoJSON files from query parameters.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

https://github.com/geosolutions-it/MapStore2/issues/8927

**What is the new behavior?**
GeoJSON can be imported with an action query parameter.
```
https://dev-mapstore.geosolutionsgroup.com/mapstore/#/viewer/config?actions=[{"type":"CATALOG:ADD_LAYERS_FROM_CATALOGS","layers":["LayerTitle"],"sources":[{"type":"GEOJSON","url":"https://pkgstore.datahub.io/examples/geojson-tutorial/example/data/db696b3bf628d9a273ca9907adcea5c9/example.geojson"}]}]
```

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: docs/developer-guide/map-query-parameters.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ Requirements:
 - The number of layers should match the number of sources
 - The source name can be a string that must match a catalog service name present in the map or an object that defines an external catalog (see example)
 
-Supported layer types are WMS, WMTS, WFS and 3D Tiles.
+Supported layer types are WMS, WMTS, WFS, 3D Tiles and GeoJSON.
 
 Example:
```
Comment (allyoucanmap) [inline docs/developer-guide/map-query-parameters.md:348]: We could add a section at the bottom to explain how to add a GeoJSON via catalog query param, eg:

---

It is possible to add GeoJSON layer using the following configuration:

```json
{
    "type": "CATALOG:ADD_LAYERS_FROM_CATALOGS",
    "layers": ["My GeoJSON Layer"],
    "sources": [{ "type":"GEOJSON", "url":"https://example.com/example.geojson" }]
}
```

`#/viewer/config?actions=[{"type":"CATALOG:ADD_LAYERS_FROM_CATALOGS","layers":["My GeoJSON Layer"],"sources":[{"type":"GEOJSON","url":"https://example.com/example.geojson"}]}]`

This GeoJSON catalog will return a single record similar to the 3D Tiles catalog and for this reason the name used in the layers array will be used to apply the title to the added vector layer.

**Review by allyoucanmap — CHANGES_REQUESTED** [review]
The implementation looks good, thanks @f-necas for the contribution!
Just some general changes below in particular a fix for the missing notification error when the url of a GeoJSON is not valid

**Review by allyoucanmap — APPROVED** [review]

**Comment by f-necas** [comment]
Thanks @allyoucanmap for the review. I've made the changes.

**Comment by allyoucanmap** [comment]
@f-necas there was a temporary problem with the readthedocs so I'll need to close and reopen the PR again to trigger the workflow

**Comment by landryb** [comment]
fwiw, went to try this with the link in the PR description, and here it fails:
```
Error
Some records have not been found: LayerTitle Please check the query URL and its parameters.
```
is the syntax in the sample url valid ? is the geosolutions dev instance deployed with this PR ?

**Comment by landryb** [comment]
right. ***blunders*** damn CORS...

**Pull request merged**